core/src/main/scala/org/apache/spark/rdd/ZippedPartitionsRDD.scala

```diff
@@ -95,6 +99,18 @@ private[spark] class ZippedPartitionsRDD2[A: ClassTag, B: ClassTag, V: ClassTag]
     rdd2 = null
     f = null
   }
+
+  private def isRandomOrder(rdd: RDD[_]): Boolean = {
+    rdd.computingRandomLevel == RDD.RandomLevel.UNORDERED
```

Comment:
Instead of this, check if order is not IDEMPOTENT.
In both unordered and indeterminate case, we should return indeterminate for this zip rdd from computingRandomLevel.

Reply:
`isRandomOrder` gives the impression that both cases (INDETERMINATE and UNORDERED) are 'random'.